packages/lesswrong/components/posts/PostFormSecondaryGroups.tsx

```diff
@@ -0,0 +1,802 @@
+      padding: 6,
+     
+    },
+    borderRadius: 2,
+  },
+  fieldWrapper: {
+    marginTop: theme.spacing.unit * 2,
+    marginBottom: theme.spacing.unit * 2,
+  },
+  secondaryOptionLabelActive: {
+    borderBottom: `none`,
+    marginBottom: -5,
+    fontSize: 15,
+    paddingBottom: 0,
+    backgroundColor: theme.palette.background.pageActiveAreaBackground,
+    color: theme.palette.text.normal,
+  },
+  formGroup: {
+    border: theme.palette.border.grey300,
+    borderRadius: 2,
+    padding: 24,
+    paddingTop: 12,
+    width: "100%",
+    ...commentBodyStyles(theme),
+    marginTop: 4,
+    [theme.breakpoints.down('xs')]: {
+      marginTop: 4,
+    },
+  },
+  formGroupTitle: {
+    ...theme.typography.commentStyle,
+    marginTop: 0,
+    marginBottom: 24
+  },
+  highlightGroup: {
+    padding: 4,
+    paddingTop: 24,
+    // TODO: figure out how to reduce height of this element
+  }
+}));
+
+function getFooterTagListPostInfo(post: EditablePost) {
+  const {
+    _id, tags, postCategory, isEvent,
+    curatedDate = null,
+    frontpageDate = null,
+    reviewedByUserId = null,
+  } = post;
+
+  return {
+    _id,
+    tags,
+    curatedDate: curatedDate?.toISOString() ?? null,
+    frontpageDate: frontpageDate?.toISOString() ?? null,
+    reviewedByUserId,
+    postCategory,
+    isEvent: isEvent ?? false,
+  };
+}
+
+function userCanEditCrosspostSettings(user: UsersCurrent | null, document: OwnableDocument) {
+  return userIsAdmin(user) || allOf(userOwns, userPassesCrosspostingKarmaThreshold)(user, document);
+}
+
+function getVotingSystemOptions(user: UsersCurrent | null) {
+  const votingSystems = getVotingSystems();
+
+  const filteredVotingSystems = user?.isAdmin
+    ? votingSystems
+    : votingSystems.filter((votingSystem) => votingSystem.userCanActivate?.());
+
+  return filteredVotingSystems.map((votingSystem) => ({
+    label: votingSystem.description,
+    value: votingSystem.name,
+  }));
+}
+
+const STICKY_PRIORITIES = {
+  1: "Low",
+  2: "Normal",
+  3: "Elevated",
+  4: "Max",
+};
+
+const PostFormSecondaryGroups = ({
+  form,
+  initialData,
+  formType,
+  currentUser,
+  addOnSubmitCallbackCustom,
+  addOnSuccessCallbackCustom,
+  addOnSubmitCallbackModerationGuidelines,
+  addOnSuccessCallbackModerationGuidelines,
+}: {
+  form: TypedReactFormApi<EditablePost, PostSubmitMeta>;
+  initialData: EditablePost;
+  formType: 'new' | 'edit';
+  currentUser: UsersCurrent | null;
+  addOnSubmitCallbackCustom: ReturnType<typeof useEditorFormCallbacks<PostsEditMutationFragment>>['addOnSubmitCallback'];
+  addOnSuccessCallbackCustom: ReturnType<typeof useEditorFormCallbacks<PostsEditMutationFragment>>['addOnSuccessCallback'];
+  addOnSubmitCallbackModerationGuidelines: ReturnType<typeof useEditorFormCallbacks<PostsEditMutationFragment>>['addOnSubmitCallback'];
+  addOnSuccessCallbackModerationGuidelines: ReturnType<typeof useEditorFormCallbacks<PostsEditMutationFragment>>['addOnSuccessCallback'];
```

Comment:
The thing to do here is to is to explicitly type those functions in EditorFormComponent (`type AddOnSubmitCallback<Fragment> = (cb: EditorSubmitCallback<Fragment> => () => void;`, similar for the other one) and then type these as `AddOnSubmitCallback<PostsEditMutationFragment>`